nmpolicy/internal/resolver/filter.go

```diff
@@ -0,0 +1,49 @@
+package resolver
+
+import (
+	"fmt"
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/ast"
+)
+
+func filter(inputState map[string]interface{}, path ast.VariadicOperator, filteredValue ast.Node) (map[string]interface{}, error) {
+	filtered, err := applyFuncOnPath(inputState, path, func(valueToFilter map[string]interface{}, key string) (interface{}, error) {
+		valueToCompare, ok := valueToFilter[key]
+		if !ok {
+			return nil, fmt.Errorf("invalid path: key %s doesnt exist in %v", key, valueToFilter)
+		}
+		match, err := matchFieldValue(valueToCompare, filteredValue)
+		if err != nil {
+			return nil, fmt.Errorf("Filter: error matching the field value : %v", err)
```

Comment:
Let's be consistent `invalid filter: ...` we can create an errors.go file with strucs like `InvalidPathError` and `InvalidFilterError` that will have the proper string and we pass the msg.

Reply:
done